examples/provisioner/100-cpu-limit.yaml

```diff
@@ -9,8 +9,15 @@ spec:
   limits:
     resources:
       cpu: 100
-  provider:
-    subnetSelector:
-      karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
-    securityGroupSelector:
-      karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  providerRef:
+    name: my-provider
+---
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: my-provider
+spec:
+  subnetSelector:
+    karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
+  securityGroupSelector:
+    karpenter.sh/discovery: "${CLUSTER_NAME}" # replace with your cluster name
```

Comment:
Can you make sure all your files have new lines at the end? I see a couple that don't.